### ISSUE TITLE
Deep linking routes to the Logged In Page

### ISSUE THREAD

**Issue by erikgoins** [FlutterFlow/flutterflow-issues#453]
When a user clicks a link, they're often routed to the "Logged In" page immediately after landing on the linked page.

In the video:
1. Shared a link to a specific event page
2. Clicked the link
3. Routed to the correct page
4. was quickly routed to the Logged In page (home)

I repeated this several times with different screens and having the app open or closed. 
https://www.loom.com/share/71b67dc319834a09aa544a2dc5e7afd4

**Comment by erikgoins** [comment]
**Expected behavior (required)**
When a user clicks a deep link URL, they should route to the URL page. 

**Current behavior (required)**
when a user clicks a deep link URL, they route to the page then immediately navigate to the "Logged In" landing page.

**To Reproduce (required)**
Generate a deep link URL. Click the link (native iOS), you'll land on the correct page then quickly be routed to the logged in page.

**Context (required)**
Effectively negates deeplinking as a feature. Users can't stay on the specific page they're trying to reach (a specific chat, post, or event). 

We've heard of issues from clients where they can't stay on the proper conversation for chat notifications as well. This means they:
1. Click the notification that they've received a new message
2. Navigate to the conversation
3. Immediately navigate to the home screen
4. Can't navigate back and don't know what conversation they were on because it's now marked as read and the notification is gone

**Screenshots / recordings**
https://www.loom.com/share/71b67dc319834a09aa544a2dc5e7afd4

**Your environment**
Testflight iOS

**Bug Report Code:**
Version of FlutterFlow used: v3.1
Platform (e.g. Web, MacOS Desktop): iOS (testflight)

**Comment by hariprasadms** [comment]
Hi @erikgoins  - Thanks for updating the issue as per issue template. Could you please provide bug report code of the project?

**Comment by davidflutterflow** [comment]
Here is the project id: gig-box-jssgcd

**Comment by cskonstantyn** [comment]
We actually experience the same issue. 

Expected behavior (required)
When a user clicks a deep link URL, they should route to the URL page.

Current behavior (required)
when a user clicks a deep link URL, they route to the page and then immediately navigate to the "Logged In" landing page.


Our issue is beyond deep links though. 

Expected behavior (required)
Pressing the button with the action Launch URL using the variable and Getting the link from API call. In other words when user presses the button would open url from the API. 

Current behavior (required)
When pressing the button the button gets the URL and leads the user to the log-in page, not the URL that was fetched from API and passed in the variable. In other words instead of getting to the url user need to get the user stuck in the app and cant get to the needed url. The problem appears on the web version.

In addition, when link from the API was passed to firestore to store it, and when try to copy directly from the firestore collection and paste this in the search, the link leads to the app and not to the pass it saying. EG: link is www.google.com when inserted in google it redirects to https://myproject.flutterflow.app 

Context (required)

Ability to navigate to the deep links provided and not push the users to the logged-in page. Our clients need to navigate to specific job posts or specific profile pages, instead they are redirected to logged in page. In addition, getting url from API call become uselsee if the link converts to route that lead to logged in page. We try to dynamically let user open external pages and instead they are being led to our own app.


Will be happy to provide any more info

**Comment by cskonstantyn** [comment]
Is there any follow-up on this issue?

**Comment by cskonstantyn** [comment]
@agreaves thank youuu! you guys are the best team ever

**Comment by agreaves** [comment]
@cskonstantyn Would you mind also generating a bug report code and would it be ok for me to test the issue / possible fix within your project? It will be very useful for confirming whether these projects are affected by the same issue (and fixed by the same solution). Thanks!

**Comment by cskonstantyn** [comment]
Yes, I am absolutely fine with this. I am not a 100% how to generate a bug report code

**Comment by agreaves** [comment]
Great, thanks! Just right click any widget in the Widget Tree and click "Get Bug Report Code"
<img width="326" alt="image" src="https://user-images.githubusercontent.com/7757399/223315348-360277d9-fd76-4700-bbb7-43550373b301.png">

**Comment by cskonstantyn** [comment]
Got it @agreaves , ITESz/Hqx89godRH7rqIKfl6hQAkQzM8bYJMl9dTeA8YNZ+rB4kAe+r+VHFVXtfgeAlEA1D9uD4e+eDKhdnLK8I6BwSuUoNo1qt6aTrjQkSWRL7QBqqOfkRBLJpXfn6g0rTRkwlSWcR1SlFkw3z/cq3qNleeY8aSfxBlZ7vfcPo=

**Comment by cskonstantyn** [comment]
@agreaves Thank youuu, you are the best! =))))) Will try it out asap =)
